professors.py

```diff
@@ -4,39 +4,32 @@
 import json
 
 class Professors:
-	def GET(self):
-		model.insert_view(web.ctx.host + web.ctx.fullpath, web.ctx.status, web.ctx.ip, web.ctx.env['HTTP_USER_AGENT'] if 'HTTP_USER_AGENT' in web.ctx.env else None, "GET")
-		web.header('Content-type', 'application/json')
-                web.header('Access-Control-Allow-Origin', '*')
+    def GET(self):
+        model.insert_view(web.ctx.host + web.ctx.fullpath, web.ctx.status, web.ctx.ip, web.ctx.env['HTTP_USER_AGENT'] if 'HTTP_USER_AGENT' in web.ctx.env else None, "GET")
+        web.header('Content-type', 'application/json')
+        web.header('Access-Control-Allow-Origin', '*')
 
-		data = web.input()
+        data = web.input()
 
-		limit = utilities.get_limit(data)
-		offset = utilities.get_offset(data)
-		type_ = ""
-		reviews = False
+        limit = utilities.get_limit(data, 100, 1000)
+        offset = utilities.get_offset(data)
+        type_ = ""
+        reviews = False
 
-		if not utilities.is_int(limit):
```

Comment:
I think by removing these lines there's an internal server error when `limit` is passed. This happens in each endpoint where limit can be a parameter.

Reply:
I thought this would be covered by the `utilities.get_limit` call above, since it checks if it's an int. Which it does, but then never raises an actual error. So I fixed that instead.